src/nexus_location/nexus_location_test.go

```
@@ -104,15 +104,17 @@ func nexusLocality(replicas int, local bool) {
 func remotelyProvisionedVolume(replicas int, local bool) {
 	deferredAssert := e2e_config.GetConfig().DeferredAssert
 
-	logf.Log.Info("remote volume")
-	nodes, err := k8stest.GetMayastorNodeNames()
-	Expect(err).To(BeNil(), "failed to retrieve list of nodes")
-	Expect(len(nodes)).ToNot(BeZero(), "no nodes found")
-	nodeName := nodes[0]
-
-	logf.Log.Info("De-scheduling mayastor on", "node", nodeName)
-	k8stest.UnlabelNode(nodeName, common.MayastorEngineLabel)
-	logf.Log.Info("Scheduling consumer pod on", "node", nodeName)
+	nodeName := ""
+	nodes, err := k8stest.GetNodeLocs()
+	Expect(err).ToNot(HaveOccurred())
+
+	for _, node := range nodes {
+		if node.MasterNode {
+			nodeName = node.NodeName
+			break
+		}
+	}
+
```

If we are scheduling the pod on the master node, we should first assert that a pod can run on the master node using a SC without local set to true.
Only then proceed with the test for using a SC with local set to true.
Otherwise we run the risk of masking failures - if any.
Please add a log statement indicating which node the pod is scheduled on.

---

Fixed, can you please take another look.